### ISSUE TITLE
Multi-AZ deployments fail to render template for spec.index

### ISSUE THREAD

**Issue by daviddob** [cloudfoundry-incubator/cf-operator#813]
This bug was encountered while deploying KubeCF 0.2.0 with CF-Operator v2.0.0-0.g0142d1e9

When multi_az is enabled, the pods are created with affinity rules for failure-domain.beta.kubernetes.io/zone=z1 where the nodes in aws have similar labels to failure-domain.beta.kubernetes.io/zone=us-west-2a . I tested changing the underlying cf_deployment manifest to point at the aws az values and this successfully gets the pods scheduled in k8s.

However, once the pods are scheduled, they all go into crashloopbackoff state during init containers. The failing init container in question is template-render and the only associated logs are

`2020/02/06 22:12:40 no job instance found for spec index '4'`
(The index changes depending on pod queried i.e. kubecf-uaa-z0-0 = 0, kubecf-uaa-z1-0 = 2, kubecf-uaa-z3-0 = 4, etc). 

It is also worth noting that the number of pods created initially do not correspond to the number of instances specified, but seems to be #AZ * #Instances

The values used for the KubeCF chart were as follows deployed to an EKS cluster created with `eksctl` 
```
Client Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.2", GitCommit:"59603c6e503c87169aea6106f57b9f242f64df89", GitTreeState:"clean", BuildDate:"2020-01-23T14:21:54Z", GoVersion:"go1.13.6", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"14+", GitVersion:"v1.14.9-eks-c0eccc", GitCommit:"c0eccca51d7500bb03b2f163dd8d534ffeb2f7a2", GitTreeState:"clean", BuildDate:"2019-12-22T23:14:11Z", GoVersion:"go1.12.12", Compiler:"gc", Platform:"linux/amd64"}
```

```
system_domain: system.eks.example-domain.com

kube:
    service_cluster_ip_range: 10.100.0.0/16
    pod_cluster_ip_range: 192.168.0.0/16

multi_az: true
high_availability: true

features:
    eirini:
        enabled: true
    autoscaler:
        enabled: false
```

Attached are statefulset, and pod yaml outputs for further debugging.
[get-pods.yml.txt](https://github.com/cloudfoundry-incubator/cf-operator/files/4171508/get-pods.yml.txt)
[kubecf-uaa-z1.yml.txt](https://github.com/cloudfoundry-incubator/cf-operator/files/4171509/kubecf-uaa-z1.yml.txt)

**Comment by manno** [comment]
I added two blockers to this bug in tracker:

* https://www.pivotaltracker.com/story/show/169727962
* https://www.pivotaltracker.com/story/show/171304400

**Comment by rohitsakala** [comment]
Hi David, Sorry for the delay.

The issue should be fixed by #843 . I tested it on kind with multiple nodes. Re-open the issue if you still face an issue.
